### ISSUE TITLE
Bug: (`id-match` with specific regex can disallow native classes)

### ISSUE THREAD

**Issue by mlwyatt** [eslint/eslint#15395]
### Environment

Node version: v16.13.0
npm version: v8.1.0
Local ESLint version: v8.2.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.1.0
(MacOS Monterey (12.0.1) Apple M1)

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: [
    'eslint:recommended',
  ],
  env: {
    browser: true,
    node: true,
    amd: true,
    jquery: true,
    es2021: true
  },
  parserOptions: { sourceType: 'module' },
  rules: {
    'id-match': ['error', '^\\$?[a-z]+([A-Z0-9][a-z0-9]+)*$', { properties: true }],
  }
};
```
</details>

```js
const myVariable = 1;
const my_other_variable = 1;

function myFunction() {
  return 1;
}

class MyClass {
}

let a = new MyClass();
let b = {id: 1};
let c = Object.keys(b);
let d = Array.from(b);

console.log(
  myVariable,
  my_other_variable,
  myFunction,
  MyClass,
  a,
  b,
  c,
  d
);
```

### What did you expect to happen?

```
error  Identifier 'my_other_variable' does not match the pattern '^\$?[a-z]+([A-Z0-9][a-z0-9]+)*$'  id-match
error  Identifier 'MyClass' does not match the pattern '^\$?[a-z]+([A-Z0-9][a-z0-9]+)*$'            id-match
```

### What actually happened?

```
error  Identifier 'my_other_variable' does not match the pattern '^\$?[a-z]+([A-Z0-9][a-z0-9]+)*$'  id-match
error  Identifier 'MyClass' does not match the pattern '^\$?[a-z]+([A-Z0-9][a-z0-9]+)*$'            id-match
error  Identifier 'Object' does not match the pattern '^\$?[a-z]+([A-Z0-9][a-z0-9]+)*$'             id-match
error  Identifier 'Array' does not match the pattern '^\$?[a-z]+([A-Z0-9][a-z0-9]+)*$'              id-match
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I know my regex will disallow `MyClass` but I would like to be able to define a format for classes different than variables and functions.

As far as I know the typical standards are:
- camelCase for variables and functions
- PascalCase/TitleCase for classes
- SCREAMING_SNAKE_CASE for constants (and possibly globals)

**Comment by nzakas** [comment]
This does seem like a bug to me. We should probably filter out global references that are native objects. I’d like some other opinions from the team, though.

**Comment by snitin315** [comment]
Yes, we should filter out global references that are native objects. I will work on it.
